tests/test_ugrid_dataset.py

```diff
@@ -1263,3 +1265,18 @@ def test_laplace_interpolate_1d():
     actual = uda.ugrid.laplace_interpolate(direct_solve=True)
     assert isinstance(actual, xugrid.UgridDataArray)
     assert np.allclose(actual, 1.0)
+
+
+def test_ugriddataset_wrap_twice(tmp_path):
+    """
+    in issue https://github.com/Deltares/xugrid/issues/208 wrapping a ds
+    twice with UgridDataset gave unexpected behaviour. This tests ensures
```

Comment:
You should mention what the unexpected behavior is.

Reply:
Another thing: I'd argue the problem is not with open_dataset, it's with the `Ugrid2d.from_dataset()` method, which internally calls `_prepare_connectivity`. So a more specific test would test just that.

Reply:
> You should mention what the unexpected behavior is.

This was resolved in the latest commit